plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +199,467 @@ in
+      buffer_close_icon = defaultNullOpts.mkStr "" ''
+        The close icon for each buffer.
+      '';
+
+      modified_icon = defaultNullOpts.mkStr "●" ''
+        The icon indicating a buffer was modified.
+      '';
+
+      close_icon = defaultNullOpts.mkStr "" "The close icon.";
 
-      modifiedIcon = helpers.defaultNullOpts.mkStr "●" "The icon indicating a buffer was modified.";
+      close_command = defaultNullOpts.mkStr "bdelete! %d" ''
+        Command or function run when closing a buffer.
+      '';
 
-      closeIcon = helpers.defaultNullOpts.mkStr "" "The close icon.";
+      custom_filter = defaultNullOpts.mkLuaFn null ''
+        ```
+        fun(buf: number, bufnums: number[]): boolean
+        ```
 
-      closeCommand = helpers.defaultNullOpts.mkStr "bdelete! %d" "Command or function run when closing a buffer.";
+        NOTE: this will be called a lot so don't do any heavy processing here.
+      '';
 
-      leftMouseCommand = helpers.defaultNullOpts.mkStr "buffer %d" "Command or function run when clicking on a buffer.";
+      left_mouse_command = defaultNullOpts.mkStr "buffer %d" ''
+        Command or function run when clicking on a buffer.
+      '';
 
-      rightMouseCommand = helpers.defaultNullOpts.mkStr "bdelete! %d" "Command or function run when right clicking on a buffer.";
+      right_mouse_command = defaultNullOpts.mkStr "bdelete! %d" ''
+        Command or function run when right clicking on a buffer.
+      '';
 
-      middleMouseCommand = helpers.defaultNullOpts.mkStr null "Command or function run when middle clicking on a buffer.";
+      middle_mouse_command = defaultNullOpts.mkStr null ''
+        Command or function run when middle clicking on a buffer.
+      '';
 
       indicator = {
-        icon = helpers.defaultNullOpts.mkStr "▎" "icon";
+        icon = defaultNullOpts.mkStr "▎" ''
+          Indicator icon.
 
-        style = helpers.defaultNullOpts.mkEnumFirstDefault [
+          This should be omitted if indicator style is not `icon`.
+        '';
+
+        style = defaultNullOpts.mkEnumFirstDefault [
           "icon"
           "underline"
-        ] "style";
+          "none"
+        ] "Indicator style.";
       };
 
-      leftTruncMarker = helpers.defaultNullOpts.mkStr "" "left trunc marker";
-
-      rightTruncMarker = helpers.defaultNullOpts.mkStr "" "right trunc marker";
+      left_trunc_marker = defaultNullOpts.mkStr "" ''
+        Left truncation marker.
+      '';
 
-      separatorStyle = helpers.defaultNullOpts.mkEnum [
-        "slant"
-        "padded_slant"
-        "slope"
-        "padded_slope"
-        "thick"
-        "thin"
-      ] "thin" "Separator style";
+      right_trunc_marker = defaultNullOpts.mkStr "" ''
+        Right truncation marker.
+      '';
 
-      nameFormatter = helpers.defaultNullOpts.mkLuaFn null ''
+      separator_style = defaultNullOpts.mkNullableWithRaw (
```

Comment:
We don't need _both_ `mkNullableWithRaw` _and_ `oneOf [ rawLua ]`.

Either use `mkNullable` with `oneOf`, or use `mkNullableWithRaw` with `either`